Storage/SQL/BrowserTable.swift

```diff
@@ -395,7 +398,8 @@ open class BrowserTable: Table {
     ", folderName" +
     ", null AS faviconID" +
     ", 1 AS is_overridden" +
-    " FROM \(TableBookmarksBuffer) WHERE is_deleted IS 0"
+    " FROM \(TableBookmarksBuffer) WHERE is_deleted IS 0" +
```

Comment:
There are two consumers of `ViewBookmarksBufferOnMirror`: the UI code (`getURLForKeywordSearch`, `valueView` in the model) and the validation code (e.g., `allBufferStructuresReferToRecords`). The latter wants the unfiltered buffer; the former wants to exclude deletions. You need a second view.